v2/event/event_unmarshal.go

```diff
@@ -96,37 +106,45 @@ func readJsonFromIterator(out *Event, iterator *jsoniter.Iterator) error {
 			// Check proper specversion
 			switch sv {
 			case CloudEventsVersionV1:
-				out.Context = &EventContextV1{}
+				out.Context = &EventContextV1{
+					ID:              id,
+					Type:            typ,
+					Source:          source,
+					Subject:         subject,
+					Time:            time,
+					DataContentType: datacontenttype,
+				}
 				appendFlag(&state, specVersionV1Flag)
 			case CloudEventsVersionV03:
-				out.Context = &EventContextV03{}
+				out.Context = &EventContextV03{
+					ID:              id,
+					Type:            typ,
+					Source:          source,
+					Subject:         subject,
+					Time:            time,
+					DataContentType: datacontenttype,
+				}
 				appendFlag(&state, specVersionV03Flag)
 			default:
 				return ValidationError{"specversion": errors.New("unknown value: " + sv)}
 			}
 
+			// Apply all extensions to the context object.
+			for key, val := range extensions {
+				if err := out.Context.SetExtension(key, val); err != nil {
```

Comment:
Because you do this, you don't need the nil check of the extensions map below anymore

Reply:
I don't think we need the nil check anyway.